### ISSUE TITLE
darwin: importing numpy fails because libblas.dylib cannot be found

### ISSUE THREAD

**Issue by smancill** [NixOS/nixpkgs#136256]
### Describe the bug
Cannot import `numpy` on Darwin because `dlopen` cannot find `libblas.dylib`.

```
dlopen(/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so, 2): Library not loaded: libblas.dylib
```

### Steps To Reproduce

On nixpkgs `master`, but it fails on 20.09 and 21.05 too (does nobody uses NumPy on Darwin?):

```
$ nix-shell -I nixpkgs=$(pwd) -p python3Packages.numpy --run 'python -c "import numpy"'
Traceback (most recent call last):
  File "/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/__init__.py", line 22, in <module>
    from . import multiarray
  File "/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/multiarray.py", line 12, in <module>
    from . import overrides
  File "/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: dlopen(/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so, 2): Library not loaded: libblas.dylib
  Referenced from: /nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so
  Reason: image not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/__init__.py", line 145, in <module>
    from . import core
  File "/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.9 from "/nix/store/amn854jw94m1cph654w91pfyvz5r5a6h-python3-3.9.6/bin/python"
  * The NumPy version is: "1.20.3"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: dlopen(/nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so, 2): Library not loaded: libblas.dylib
  Referenced from: /nix/store/g3csla2ac9pvsadsvrpwh5gywpfki88x-python3.9-numpy-1.20.3/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-darwin.so
  Reason: image not found
```

The workaround in to set this on `shell.nix`, but it is annoying to remember it:

```
DYLD_FALLBACK_LIBRARY_PATH = "${numpy.blas}/lib";
```

### Expected behavior

Importing `numpy` doesn't fail.

### Notify maintainers

@fridh 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.4.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(smancill): `"nixpkgs-21.05pre294098.50d17a46639, unstable-21.11pre310022.14b0f20fa1f"`
 - nixpkgs: `/Users/smancill/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- python3Packages.numpy
```

**Comment by smancill** [comment]
This looks like a `blas` problem. The _install name` of the `libblas` library is not set to an absolute path, which should be the reason the loader cannot load it.

https://github.com/NixOS/nixpkgs/blob/977ac07fed14848c0dcbf0e17da1f5ea2778d600/pkgs/build-support/alternatives/blas/default.nix#L86

**Comment by smancill** [comment]
That wrong `blas` _install name_ should also be the reason for #107747.

I will try fixing it and recompiling NumPy and see.

**Comment by smancill** [comment]
It works:

```
$ nix-shell -I nixpkgs=$(pwd) -p python3Packages.numpy --run 'python -c "import numpy; print(numpy.__version__)"'
1.20.3
```

**Comment by veprbl** [comment]
Should be fixed once #135493 hits the master

**Comment by smancill** [comment]
I just did the same. Didn't think about searching the PRs first.

I can confirm that it works too.

**Comment by smancill** [comment]
Fixed now that both #135493 and #136535 landed on master.
